include/deal.II/base/parameter_acceptor.h

```diff
@@ -373,29 +373,40 @@ class ParameterAcceptor : public Subscriptor
    * default values, and don't want to read external files to use a class
    * derived from ParameterAcceptor.
    *
-   * If outfilename is not the empty string, then write the content that was
-   * read in to the outfilename. The format of both input and output files are
-   * selected using the extensions of the files themselves. This can be either
-   * `prm` or `xml` for input, and `prm`, `xml`, or `tex/latex` for output. If
-   * the output format is `prm`, then `output_style_for_prm_format` is used to
-   * decide whether we write the full documentation as well, or only the
-   * parameters.
+   * If @p output_filename is not the empty string, then we write the content
+   * that was read into the @p output_filename file, using the style specified
+   * in @p output_style_for_output_filename. The format of both input and output
+   * files are selected using the extensions of the files themselves. This can
+   * be either `prm`, `xml`, or `json` for the @p filename, and any of the
+   * supported formats for the @p output_filename.
    *
    * If the input file does not exist, a default one with the same name is
-   * created for you, and an exception is thrown.
+   * created for you following the style specified in
+   * @p output_style_for_filename, and an exception is thrown.
+   *
+   * By default, the file format used to write the files is deduced from
+   * the extension of the file names, ignoring the corresponding
+   * ParameterHandler::OutputStyle specification. The corresponding
```

Comment:
That's confusing. Why not:
* If `output_style` doesn't have any of the file-type bits set, use the output format that corresponds to the extension only 
* Else, if the file extension is recognized, assert that it matches the bit in `output_style`
* Else, use the bit from the `output_style`
?